clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3528,7 +3528,9 @@ class SYCLKernelNameTypeVisitor
           // case will resolve to ::main::KernelName1 (instead of
           // ::KernelName1). Since this is not visible to runtime code that
           // submits kernels, this is invalid.
-          if (Tag->isCompleteDefinition() || S.getLangOpts().SYCLIsHost) {
+          if (Tag->isCompleteDefinition() ||
+              (S.getLangOpts().SYCLIsHost &&
```

Comment:
Are both conditions here required?  Does SYCLEnableIntHeader not imply SYCLIsHost?  I would think we would want to error-out during cc1 parsing if BOTH aren't passed together.

Reply:
> Does SYCLEnableIntHeader not imply SYCLIsHost?

I suppose It does.

> I would think we would want to error-out during cc1 parsing if BOTH aren't passed together.

We can emit an error if `SYCLEnableIntHeaderDiags` is true without `SYCLIsHost` being true. The opposite can happen in valid situations, though. And this patch actually is an attempt to fix spurious errors happening due to https://github.com/intel/llvm/pull/4945 when -fsycl-is-host is passed alone.

Reply:
And it is done.

Reply:
>>We can emit an error if SYCLEnableIntHeaderDiags is true without SYCLIsHost being true. The opposite can happen in valid situations, though

Right, yeah, this is what I'd meant.